src/main/java/org/opensearch/security/configuration/ClusterInfoHolder.java

```diff
@@ -26,49 +26,23 @@
 
 package org.opensearch.security.configuration;
 
-import java.util.Iterator;
-import java.util.List;
-
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import org.opensearch.LegacyESVersion;
 import org.opensearch.cluster.ClusterChangedEvent;
-import org.opensearch.cluster.ClusterState;
 import org.opensearch.cluster.ClusterStateListener;
-import org.opensearch.cluster.metadata.IndexMetadata;
 import org.opensearch.cluster.node.DiscoveryNode;
 import org.opensearch.cluster.node.DiscoveryNodes;
-import org.opensearch.index.Index;
 
 public class ClusterInfoHolder implements ClusterStateListener {
 
     protected final Logger log = LogManager.getLogger(this.getClass());
-    private volatile Boolean has6xNodes = null;
-    private volatile Boolean has6xIndices = null;
     private volatile DiscoveryNodes nodes = null;
     private volatile Boolean isLocalNodeElectedClusterManager = null;
     private volatile boolean initialized;
     
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
-        final boolean isTraceEnabled = log.isTraceEnabled();
-        if(has6xNodes == null || event.nodesChanged()) {
-            has6xNodes = Boolean.valueOf(clusterHas6xNodes(event.state()));
-            if (isTraceEnabled) {
-                log.trace("has6xNodes: {}", has6xNodes);
-            }
-        }
-        
-        final List<String> indicesCreated = event.indicesCreated();
-        final List<Index> indicesDeleted = event.indicesDeleted();
-        if(has6xIndices == null || !indicesCreated.isEmpty() || !indicesDeleted.isEmpty()) {
-            has6xIndices = Boolean.valueOf(clusterHas6xIndices(event.state()));
-            if (isTraceEnabled) {
-                log.trace("has6xIndices: {}", has6xIndices);
-            }
-        }
-        
         if(nodes == null || event.nodesChanged()) {
```

Comment:
We only need to track when nodes change now not also when indices are created & deleted?

Reply:
The purpose of this code is to log if there are 6.X nodes or indices in the cluster upon change (node was added/removed from the cluster or an index was created or deleted). If its not possible to run es 6.X nodes with os 3.X then this code will never be used.